lib/utils/ast-utils.js

```diff
@@ -149,8 +150,8 @@ function createIdentifierOrLiteral(j, val) {
 			return createExternalRegExp(j, val);
 		} else {
 			// Use identifier instead
-			// TODO: Check if literalVal is a valid Identifier!
-			return j.identifier(literalVal);
+			if (!literalVal.match(generateIdentifierRegex()))
```

Comment:
The regex will be generated each time this branch is visited. Maybe we could generate it upfront?

Reply:
@ooflorent the generated regex is 14971 characters long and it is not a good practice to use such a huge regex directly also esprima, Acorn etc. validates identifier this way hence I thought it will be right to do not use it directly as it will be difficult to debug and modify codinghorror [blog](https://blog.codinghorror.com/regex-use-vs-regex-abuse/) mentioned it. Apart from this, I have replaced `match` with `test` to increase the [performance](https://jsperf.com/regexp-test-vs-match-m5) but still, it takes too much time and the build fails.Thanks !!